### PR TITLE
D8CORE-2968: Fix for deleted schedule items still appearing

### PR DESCRIPTION
**Pull request by imonroe**
# READY FOR REVIEW

# Summary
- Event Schedule items were still showing up on the page, even though they had been removed from the Event.
- We fix this by modifying the view which generates the schedule list, and do an inner join with the `node__su_event_schedule` table, which allows us to exclude Schedule items which have been removed, but which still have revisions in the database.

# Review By (Date)
- Today, hopefully.  Needed for 2.0.0

# Criticality
- 5
- affects all sites that use event schedule paragraphs.

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Create an Event.
3. Add a Schedule to the event.
4. Note it appears correctly on the page.
5. Edit the Event.
6. Remove the Schedule paragraph.
7. Save the page.
8. Notice that the deleted Schedule item no longer appears on the page.

### Code
- [X] Are the naming conventions following our standards?
- [X] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [X] Are tests provided? eg (unit, behat, or codeception)

### Code security
- n/a

## General
- [X] Is there anything included in this PR that is not related to the problem it is trying to solve?
  - A small adjustment to the composer.json to reflect the dependency on entity_reference_revisions which was previously missing.
- [ ] Is the approach to the problem appropriate?

# Associated Issues and/or People
- D8CORE-2968

### CODE REVIEW

**Comment by imonroe** [comment]
My tests pass, but it's a hassle because they seem to either pass on d9 and fail on d8, or pass on d8 and fail on d9.  This is the random-test-failure issue that's been happening in this repo.

File: stanford_events.module [hunk 1]
```diff
@@ -131,6 +131,22 @@ function stanford_events_views_post_render(ViewExecutable $view, &$output, Cache
 
 }
 
+/**
+ * Implements hook_views_pre_execute().
+ */
+function stanford_events_views_pre_execute(ViewExecutable $view) {
```
Comment (pookmish) [inline stanford_events.module:137]: This seems pretty heavy handed and will introduce more obscurity than it should. Perhaps we should investigate an alternative solution like rebuilding the view to satisfy the end result.

Comment (imonroe) [reply stanford_events.module:137]: I'm certainly open to that; I tried rebuilding the view in several ways, but when you build a view that lists paragraphs through entity_reference_revisions, it uses the `paragraphs_item_field_data` table as the base table for the view, and that table will include revisions which are no longer referenced on the Event node.

If you've got a more elegant approach, I'd be happy to implement it, but this is the least intrusive way I could find. 🤷

Comment (pookmish) [reply stanford_events.module:137]: https://github.com/SU-SWS/stanford_profile/pull/354

Comment (pookmish) [reply stanford_events.module:137]: Styles and some small tweaks probably still need to be done though.

**Review by pookmish — APPROVED** [review]

File: tests/codeception/functional/StanfordEventsScheduleCest.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      $I->logInWithRole('administrator');
+        $path = $this->node->toUrl()->toString();
+        $id = $this->node->id();
+
+        $I->amOnPage("node/$id/edit");
+        $I->canSee("Schedule Details");
+        $I->click("Add Schedule");
+        $I->wait(3);
+        $I->seeElement('input', ['name' =>  'su_event_schedule[0][subform][su_schedule_headline][0][value]']);
+        $I->fillField(['name' => 'su_event_schedule[0][subform][su_schedule_date_time][0][value][date]'], '12012021');
```
Comment (pookmish) [inline tests/codeception/functional/StanfordEventsScheduleCest.php:56]: fine for now, but next time try to use the normal field labels to make these tests more human readable.

**Pull request merged**